vcd/resource_vcd_vapp_org_network_test.go

```diff
@@ -0,0 +1,202 @@
+		return
+	}
+
+	resourceName = "TestAccVcdVappOrgNetwork_Fenced"
+
+	var params = StringMap{
+		"Org":                         testConfig.VCD.Org,
+		"Vdc":                         testConfig.VCD.Vdc,
+		"resourceName":                resourceName,
+		"vappName":                    vappNameForNetworkTest,
+		"EdgeGateway":                 testConfig.Networking.EdgeGateway,
+		"NetworkName":                 "TestAccVcdVAppNetFenced",
+		"orgNetwork":                  "TestAccVcdVAppNetFenced",
+		"firewallEnabled":             "false",
+		"firewallEnabledForUpdate":    "true",
+		"natEnabled":                  "false",
+		"natEnabledForUpdate":         "true",
+		"retainIpMacEnabled":          "true",
+		"retainIpMacEnabledForUpdate": "false",
+		"isFenced":                    "true",
+		"isFencedForUpdate":           "true",
+		"FuncName":                    "TestAccVcdVappOrgNetwork_Fenced",
+	}
+
+	rungVappOrgNetworkTest(t, params)
+}
+
+func rungVappOrgNetworkTest(t *testing.T, params StringMap) {
+	configText := templateFill(testAccCheckOrgVappNetwork_basic, params)
+	debugPrintf("#[DEBUG] CONFIGURATION: %s", configText)
+	params["FuncName"] = t.Name() + "-Update"
+	updateConfigText := templateFill(testAccCheckOrgVappNetwork_update, params)
+	debugPrintf("#[DEBUG] CONFIGURATION: %s", updateConfigText)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckVappNetworkDestroy,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: configText,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckVappNetworkExists("vcd_vapp_org_network."+params["resourceName"].(string)),
+					resource.TestCheckResourceAttr(
+						"vcd_vapp_org_network."+params["resourceName"].(string), "vapp_name", params["vappName"].(string)),
+					resource.TestCheckResourceAttr(
+						"vcd_vapp_org_network."+params["resourceName"].(string), "org_network_name", params["orgNetwork"].(string)),
+					resource.TestCheckResourceAttr(
+						"vcd_vapp_org_network."+params["resourceName"].(string), "retain_ip_mac_enabled", params["retainIpMacEnabled"].(string)),
+					resource.TestCheckResourceAttr(
+						"vcd_vapp_org_network."+params["resourceName"].(string), "is_fenced", params["isFenced"].(string)),
+					resource.TestCheckResourceAttr(
+						"vcd_vapp_org_network."+params["resourceName"].(string), "firewall_enabled", params["firewallEnabled"].(string)),
+					resource.TestCheckResourceAttr(
+						"vcd_vapp_org_network."+params["resourceName"].(string), "nat_enabled", params["natEnabled"].(string)),
+				),
+			},
+			resource.TestStep{
+				Config: updateConfigText,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckVappNetworkExists("vcd_vapp_org_network."+params["resourceName"].(string)),
+					resource.TestCheckResourceAttr(
+						"vcd_vapp_org_network."+params["resourceName"].(string), "vapp_name", params["vappName"].(string)),
```

Comment:
Can we use a variable instead of `"vcd_vapp_org_network."+params["resourceName"].(string)` over and over?

Reply:
changed